### PR TITLE
Nc news 4

### PR DESCRIPTION
**Pull request by oMotevalli**
ticket 5

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -51,6 +51,37 @@ afterAll(() => {
...
+                expect(body).toBeInstanceOf(Object);
+                expect(body.article[0].article_id).toBe(5);
+            })
+        })
+        test("200: should respond with an article object containing author title, article_id, body, topic, created_at, votes, article_img_url", () => {
+            return request(app)
+            .get("/api/articles/6")
+            .expect(200)
+            .then(({ body }) => {
+                expect(body.article[0]).toMatchObject({
```
Comment (yalicee) [inline __tests__/app.test.js:69]: You can test with the actual data that article 6 has

File: __tests__/app.test.js [hunk 2]
```diff
@@ -51,6 +51,37 @@ afterAll(() => {
...
     });
 
+    describe("GET /api/articles/:article_id", () => {
+        test("200: should respond with an article object, matching the inputted article ID", () => {
+            return request(app)
+            .get("/api/articles/5")
+            .expect(200)
+            .then(({ body }) => {
+                expect(body).toBeInstanceOf(Object);
+                expect(body.article[0].article_id).toBe(5);
```
Comment (yalicee) [inline __tests__/app.test.js:61]: Article should be an object not an array if it's just singular

File: __tests__/app.test.js [hunk 3]
```diff
@@ -94,3 +126,47 @@ afterAll(() => {
...
+        test("200: should return the array of article objects in date descending order", () => {
+            return request(app)
+            .get("/api/articles")
+            .expect(200)
+            .then(({ body }) => {
+                const { articles } = body;
+                expect(articles).toBeSortedBy("created_at", { descending: true });
+              });
+        })
+        test("404: should return a 404 'not found' if incorrect endpoint specified", () => {
```
Comment (yalicee) [inline __tests__/app.test.js:161]: You should already made this test

File: models/models.js [hunk 4]
```diff
@@ -21,3 +21,26 @@ exports.selectArticleByID = (article_id) => {
...
+    return db
+    .query(`
+    SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(comments.article_id) AS comment_count
+    FROM articles
+    LEFT JOIN comments
+    ON articles.article_id = comments.article_id
+    GROUP BY articles.article_id
+    ORDER BY created_at DESC;`)
+    .then(({rows}) => {
+        console.log(rows);
```
Comment (yalicee) [inline models/models.js:35]: Remove all console.logs before you push code

File: models/models.js [hunk 5]
```diff
@@ -21,3 +21,26 @@ exports.selectArticleByID = (article_id) => {
...
+        return rows
+    })
+}
+
+
+// agg group by
```
Comment (yalicee) [inline models/models.js:41]: Remove all comments that aren't relevant

**Review by yalicee — APPROVED** [review]
Make changes in comments before you merge

File: __tests__/app.test.js (hunk 1 above)
Comment (oMotevalli) [reply __tests__/app.test.js:69]: this test has been approved and merged in a previous ticket/PR. also, it's throwing errors if  change the test.

File: __tests__/app.test.js (hunk 2 above)
Comment (oMotevalli) [reply __tests__/app.test.js:61]: also from previous ticket/PR

File: __tests__/app.test.js (hunk 3 above)
Comment (oMotevalli) [reply __tests__/app.test.js:161]: i don't understand, this is testing the current endpoint's ability to return a 404. previous 404 test was for a previous endpoint/ticket.

File: models/models.js (hunk 4 above)
Comment (oMotevalli) [reply models/models.js:35]: done

File: models/models.js (hunk 5 above)
Comment (oMotevalli) [reply models/models.js:41]: done

**Pull request merged**